src/buildstream/source.py

```diff
@@ -1078,11 +1063,8 @@ def collect_source_info(self) -> Iterable[SourceInfo]:
         SourceInfo objects created by implementors should be created with
         :func:`Source.create_source_info() <buildstream.source.Source.create_source_info>`.
 
-        Returns: the :class:`.SourceInfo` objects describing this source
-
-        Raises:
-           :class:`.SourceImplError`: if the source class does not implement this method and does not implement
-                                      :func:`SourceFether.get_source_info() <buildstream.source.SourceFetcher.get_source_info>`
+        Returns: the :class:`.SourceInfo` objects describing this source, or ``None`` if the
+                 SourceFetcher does not implement this method.
```

Comment:
I think you meant Source rather than SourceFetcher.

Reply:
Ah good catch, copy/paste error.

Fixed.